sopel/plugins/rules.py

```diff
@@ -1301,3 +1360,123 @@ def parse(self, text):
             match = regex.search(text)
             if match:
                 yield match
+
+
+class URLCallback(Rule):
+    """URL Callback rule definition.
+
+    An URL Callback rule (or simply "an URL rule") detects URLs in a trigger
+    then it uses regular expressions to match at most once per URL per regular
+    expression, i.e. you can trigger between 0 and the number of regex the URL
+    callback has per URL in the IRC line.
+
+    Here is an example with an URL rule with the pattern
+    ``r'https://example\\.com/(.*)'``:
+
+    .. code-block:: irc
+
+        <user> https://example.com/test
+        <Bot> You triggered an URL callback, with the "/test" path
+        <user> and this URL is https://example.com/other can you get it?
+        <Bot> You triggered an URL callback, with the "/other" path
+
+    Like generic rules, URL callback rules are not triggered by any specific
+    name and they don't have aliases.
+
+    .. note::
+
+        Unlike generic rules and commands, the :func:`~sopel.plugin.url`
+        decorator expect its decorated function to have the bot and the trigger
```

Comment:
```suggestion
        decorator expects its decorated function to have the bot and the trigger
```

Found the first nemesi**S** of this PR. You're getting better!